packages/itwin/tree-widget/src/test/trees/category-tree/CategoriesTreeComponent.test.tsx

```diff
@@ -27,31 +40,215 @@ describe("<CategoriesTreeComponent />", () => {
     await IModelApp.shutdown();
   });
 
+  beforeEach(async () => {
+    vpMock = mockViewport();
+    const { presentationManager } = mockPresentationManager();
+    sinon.stub(Presentation, "presentation").get(() => presentationManager.object);
+    await Presentation.initialize();
+  });
+
+  afterEach(() => {
+    imodelMock.reset();
+    sinon.restore();
+    Presentation.terminate();
+  });
+
   const imodelMock = moq.Mock.ofType<IModelConnection>();
+  const viewsMock = moq.Mock.ofType<IModelConnection.Views>();
+
+  imodelMock.setup((x) => x.views).returns(() => viewsMock.object);
   let vpMock = moq.Mock.ofType<Viewport>();
 
+  const createKey = (id: Id64String): ECInstancesNodeKey => {
+    return {
+      type: StandardNodeTypes.ECInstancesNode,
+      version: 0,
+      instanceKeys: [{ className: "MyDomain:SpatialCategory", id }],
+      pathFromRoot: [],
+    };
+  };
+
   const categories: CategoryInfo[] = [{
     categoryId: "CategoryId",
     subCategoryIds: ["SubCategoryId1", "SubCategoryId2"],
   }];
 
+  const categoryNodes: PresentationTreeNodeItem[]  = [
+    {
+      key: createKey(categories[0].categoryId),
+      id: categories[0].categoryId,
+      label: PropertyRecord.fromString("category-node"),
+      autoExpand: true,
+      hasChildren: true,
+    } as PresentationTreeNodeItem,
+    {
+      key: createKey(categories[0].categoryId),
+      id: categories[0].categoryId,
+      label: PropertyRecord.fromString("category-node"),
+      autoExpand: true,
+      hasChildren: false,
+    } as PresentationTreeNodeItem,
+  ];
+
   const filteredCategories: CategoryInfo[] = [{
     categoryId: "FilteredCategoryId",
     subCategoryIds: ["FilteredSubCategoryId1", "FilteredSubCategoryId2"],
   }];
 
-  afterEach(() => {
-    imodelMock.reset();
-    sinon.restore();
-    vpMock = mockViewport();
+  const viewport =  {
+    onDisplayStyleChanged: new BeEvent<(vp: Viewport) => void>(),
+    onViewedCategoriesChanged: new BeEvent<(vp: Viewport) => void>(),
+    view: {
+      is3d: sinon.fake.returns(false),
+    },
+  } as unknown as Viewport;
+
+  const iModel = {
+    createQueryReader: sinon.fake.returns({
+      toArray: sinon.fake.returns(new Promise((resolve) => resolve([]))),
+    }) as unknown as ECSqlReader,
+    categories: {
+      getCategoryInfo: sinon.fake.returns(new Promise((resolve) => resolve(new Map<Id64String, IModelConnection.Categories.CategoryInfo>()))),
+    } as unknown as IModelConnection.Categories,
+  } as unknown as IModelConnection;
+
+  describe("<CategoriesTreeComponent />", () => {
+
+    it("returns null if iModel is undefined", () => {
```

Comment:
I think these tests that renders `CategoryTreeComponent` without `iModel` or `viewport` should stub `CategoryTree` and verify it was not called.